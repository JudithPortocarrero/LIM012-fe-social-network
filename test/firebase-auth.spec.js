//  tarear funciones puras de firebase

import {
  loguearUser, createUser, verificationEmail,
} from '../src/firebase/firebase-conexion.js';


const firebasemock = require('firebase-mock');

const mockauth = new firebasemock.MockAuthentication();
mockauth.autoFlush();

global.firebase = firebasemock.MockFirebaseSdk(
  () => null,
  () => mockauth,
);
describe('funcion loguearUser con email y password', () => {
  it('Deberia loguearse un usuario con email y conraseña', (done) => {
    loguearUser('example@gmail.com', '123456').then((user) => {
      expect(user.email).toMatchSnapshot('example@gmail.com');
      done();
    });
  });
});
describe('funcion createUser con email y password', () => {
  it('Deberia crearse un usuario con email y conraseña', (done) => {
    createUser('exampleCreate@outlook.com', '123456').then((user) => {
      expect(user.email).toMatchSnapshot('exampleCreate@outlook.com');
      done();
    });
  });
});


describe('funcion verificationEmail con email y password', () => {
  it('Deberia enviarse un email a un usuario registrado', () => {
    expect.assertions(1);
    firebase.auth().currentUser.sendEmailVerification = verificationEmail();
  });
});
