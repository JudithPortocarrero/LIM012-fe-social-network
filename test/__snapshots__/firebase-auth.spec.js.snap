// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`funcion createUser con email y password Deberia crearse un usuario con email y conraseña: exampleCreate@outlook.com 1`] = `"exampleCreate@outlook.com"`;

exports[`funcion loguearUser con email y password Deberia loguearse un usuario con email y conraseña: example@gmail.com 1`] = `"example@gmail.com"`;
